### ISSUE TITLE
Prompt Bot Doesn't "Forget" When Restarting New Convo

### ISSUE THREAD

**Issue by Zerryth** [Microsoft/BotBuilder-Samples#783]
## Sample information
1. Sample type:  samples 25. logger-bot, 26. transcript-logger-bot, .05 multi-turn-prompt
2. Sample language: nodejs

## Bug Description & Repro
Bot lets you go through the conversation once (it welcomes, prompts  you for name, and if you want to give your age), however when you Restart conversation in emulator or close the bot and open it back up again to try to begin a new conversation, the bot does not "forget". It remembers the answers that were input and doesn't reprompt for name/age

buggy behavior:
![image](https://user-images.githubusercontent.com/35248895/47580928-01057980-d905-11e8-834b-30bcb07696a6.png)

## Expected behavior
Bot should clear name/age when starting a new conversation (i.e. if I click Restart conversation in emulator, bot should go through the steps of welcoming and asking for name and age again)

[bug]

**Comment by nyxsys** [comment]
Pasting from the teams conversation, but would like confirmation:

> I suspect that the "saved name" functionality might be intentional
> to show off being able to handle the same user across multiple conversations in a bot
> i.e if you opened up the bot in a different version of the emulator entirely or whatever you'd be able to talk to it from the start, or if you could change your username somehow it'd work
> this functionality could be clearer but I think that's what's going on
> 
> It appears that this is also how this sample functions within the C# samples repo

**Comment by Zerryth** [comment]
Yes, confirmation should be nailed down if this is actually in fact the intended behavior. 

After downloading even the v4 C# "Basic Bot" template from ABS and chatting with the bot and restarting the conversation, you'll notice that that bot also does not "forget" the previous input answers from the users and just repeats "your name is..." over and over, even after closing the window and logging back into ABS.

If this is the intended behavior for sample 05. (and in turn also 25. & 26. which are based off of the bot in .05), then we can move forward with approving review of #766.

Not sure then which person would be the best Mark & I should talk to. Looks like Ben Brown wrote the commit for sample .05 though

**Comment by benbrown** [comment]
Hello! Thank you for your insightful questions.

Yes, this is functioning as designed - these values are associated with the USER, not the CONVERSATION.  The SDK supports both, but in these samples, we've used the user state manager to capture user name, etc. 
This causes it to persist between conversations.
